### PR TITLE
generalize helper for loading module from source

### PR DESCRIPTION
**Pull request by bowenliang123**
# Description

- generalize module_import_helper for loading module from source
- it's used for avoid repeated code by
  - importing a module from the source file directly
     - Check, import and update the module name into `sys.modules`, according to  Python docs [Importing a source file directly]( https://docs.python.org/3/library/importlib.html#importing-a-source-file-directly)
     - Support to use lazy import loader (disabled by default), which creates a real module object but is not loaded into memory, according to [Implementing lazy imports](https://docs.python.org/3/library/importlib.html#implementing-lazy-imports). It may help to improve faster launch time and prevent actual loading unused modules (for LLM models, built-in tools, etc.), when it's turned on by default in the future.
   - getting subclasses of targeted parent type from the module

## Type of Change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update, included: [Dify Document](https://github.com/langgenius/dify-docs)

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] TODO

# Suggested Checklist:

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] My changes generate no new warnings
- [x] I ran `dev/reformat`(backend) and `cd web && npx lint-staged`(frontend) to appease the lint gods
- [ ] `optional` I have made corresponding changes to the documentation 
- [ ] `optional` I have added tests that prove my fix is effective or that my feature works
- [ ] `optional` New and existing unit tests pass locally with my changes

### CODE REVIEW

**Review by crazywoola — DISMISSED** [review]

File: api/core/utils/module_import_helper.py [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+        existed_spec = importlib.util.find_spec(module_name)
+        if not existed_spec:
+            spec = existed_spec
+        else:
+            # Refer to: https://docs.python.org/3/library/importlib.html#importing-a-source-file-directly
+            spec = importlib.util.spec_from_file_location(module_name, py_file_path)
+            if use_lazy_loader:
+                # Refer to: https://docs.python.org/3/library/importlib.html#implementing-lazy-imports
+                spec.loader = importlib.util.LazyLoader(spec.loader)
+        module = importlib.util.module_from_spec(spec)
```
Comment (takatost) [inline api/core/utils/module_import_helper.py:26]:
```
ERROR:root:Failed to load module openai_moderation from /Users/takatost/Projects/dify/api/core/moderation/openai_moderation/openai_moderation.py: 'NoneType' object has no attribute 'loader'
Traceback (most recent call last):
  File "/Users/takatost/Projects/dify/api/core/utils/module_import_helper.py", line 26, in import_module_from_source
    module = importlib.util.module_from_spec(spec)
  File "<frozen importlib._bootstrap>", line 568, in module_from_spec
AttributeError: 'NoneType' object has no attribute 'loader'
WARNING:root:Missing subclass of Moderation in /Users/takatost/Projects/dify/api/core/moderation/openai_moderation/openai_moderation.py, Skip.
ERROR:root:Failed to load module keywords from /Users/takatost/Projects/dify/api/core/moderation/keywords/keywords.py: 'NoneType' object has no attribute 'loader'
Traceback (most recent call last):
  File "/Users/takatost/Projects/dify/api/core/utils/module_import_helper.py", line 26, in import_module_from_source
    module = importlib.util.module_from_spec(spec)
  File "<frozen importlib._bootstrap>", line 568, in module_from_spec
AttributeError: 'NoneType' object has no attribute 'loader'
WARNING:root:Missing subclass of Moderation in /Users/takatost/Projects/dify/api/core/moderation/keywords/keywords.py, Skip.
```

load err in moderation extensions.

**Review by takatost — CHANGES_REQUESTED** [review]
some err happened when module loading

File: api/core/utils/module_import_helper.py (hunk 1 above)
Comment (bowenliang123) [reply api/core/utils/module_import_helper.py:26]: Let me check again. To see whether the same on my local env.

Comment (bowenliang123) [reply api/core/utils/module_import_helper.py:26]: Or we could just skip the potential lazy loader support.

Comment (bowenliang123) [reply api/core/utils/module_import_helper.py:26]: Hi, the error comes from setting the spec when there's no existed spec found.
```
        if existed_spec:
            spec = existed_spec
        ...
        module = importlib.util.module_from_spec(spec)
        ...
```
It has been fixed by reverting the condition.

**Comment by bowenliang123** [comment]
Tests added. Please review again.

**Review by takatost — APPROVED** [review]
LGTM

**Pull request merged**